### ISSUE TITLE
Adding documents to a VisualStudioWorkspace in a subfolder fails if folder doesn't exist

### ISSUE THREAD

**Issue by twsouthwick** [dotnet/roslyn#73783]
**Version Used**: 

**Steps to Reproduce**:

1.  Get a VS workspace
2. Create a document in a new folder:

```csharp
private void Test(Workspace ws)
{
  var project = ws.CurrentSolution.GetProject(projectId);
  project = project.AddDocument("test", SourceText.From("using System;"), ["folder"])
    .Project;
  // switch to VS main thread
  ws.TryApplyChanges(project.Solution);
}
```

**Expected Behavior**:
Creates the folder if it doesn't exist and add the file

**Actual Behavior**:
Throws "Could not find a part of the path '....'" at https://github.com/dotnet/roslyn/blob/12f89683716864af2582b59f9b94395ad8f39910/src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs#L991

**Comment by CyrusNajmabadi** [comment]
I don't believe this is supported.  This is how you tell our workspace about a file that actually exists.

**Comment by twsouthwick** [comment]
It seems like inconsistent behavior then - you can add a file that doesn't exist *if* the folder exists. So it works for files that don't exist in some cases but not others.

**Comment by CyrusNajmabadi** [comment]
Fair point.  :)  I woudl be ok with us uniformly allowing both.  Checking with @jasonmalinowski  if he has any concerns.

**Comment by jasonmalinowski** [comment]
Yeah I'd expect this to work.

**Comment by CyrusNajmabadi** [comment]
SGTM.  Thanks! :)

**Comment by CyrusNajmabadi** [comment]
@twsouthwick We'd take a contribution here if you're interested!

**Comment by twsouthwick** [comment]
Sounds good - feel free to assign it to me
